src/Microsoft.ML.AutoML/Utils/UserInputValidationUtil.cs

```diff
@@ -57,6 +57,14 @@ public static void ValidateNumberOfCVFoldsArg(uint numberOfCVFolds)
             }
         }
 
+        public static void ValidateSamplingKey(string samplingKeyColumnName, string groupIdColumnName, TaskKind task)
+        {
+            if (task == TaskKind.Ranking && samplingKeyColumnName != null && samplingKeyColumnName != groupIdColumnName)
+            {
```

Comment:
if `samplingKeyColumnName` is null, I think we'd also need to throw, since we _need_ it to be `groupId`, so it shouldn't be null.